tapdb/migrations.go

```diff
@@ -109,6 +120,31 @@ func (m *migrationLogger) Verbose() bool {
 	return m.log.Level() <= btclog.LevelDebug
 }
 
+// maxKnownMigrationVersion returns the maximum migration version known to the
+// given source driver.
+func maxKnownMigrationVersion(sourceDriver source.Driver) (fn.Option[uint],
```

Comment:
If `fn.None[uint](), nil` is never returned, why do we even need the option here? What's wrong with the well-established `return 0, err` pattern?
IMO we should only use the option type (which decreases readability) when `<none>, nil` is actually an option as a return value in a function.

Reply:
I've removed `maxKnownMigrationVersion`. But I'll keep your `return 0, err` pattern comment in mind for next time.